src/Compilers/VisualBasic/Portable/Symbols/Source/SourceNamedTypeSymbol.vb

```diff
@@ -2084,8 +2084,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
         Friend NotOverridable Overrides ReadOnly Property ObsoleteAttributeData As ObsoleteAttributeData
             Get
-                ' If there are no attributes then this symbol is not Obsolete.
-                If Not GetAttributeDeclarations().Any() Then
+                ' If there are no attributes then this symbol is not Obsolete. Only apply this optimization for cases
+                ' where we know the syntax tree is loaded, or we could force a complete tree deserialization (and thus
+                ' no longer be an early return optimization).
+                If Not IsPartial AndAlso Not GetAttributeDeclarations().Any() Then
```

Comment:
>Not IsPartial AndAlso Not GetAttributeDeclarations().Any() [](start = 19, length = 58)

I think we shouldn't suppress this optimization for partial types, instead we should add a HasAnyAttributes property (which wouldn't need syntax trees alive) to MergedTypeDeclaration and utilize it here. Or we can replace this ```If``` with an inline check the same way C# implementation does it.